lib/3scale_toolbox/commands/copy_command/copy_service.rb

```diff
@@ -54,13 +54,26 @@ def self.endpoint_from_url(url)
           url.sub /\w*@/, ''
         end
 
+        def self.service_valid_params
+          %w[
+            name backend_version deployment_option description
+            system_name end_user_registration_required
+            support_email tech_support_email admin_support_email
+          ]
+        end
+
+        # Returns new hash object with not nil valid params
+        def self.filter_params(valid_params, source)
+          valid_params.each_with_object({}) do |key, target|
+            target[key] = source[key] unless source[key].nil?
```

Comment:
What about:

```ruby
target[key] = source[key] if source.key?(key)
```

Reply:
key needs to exist and the value not nil